addon/mixins/data-adapter-mixin.js

```diff
@@ -94,8 +95,10 @@ export default Ember.Mixin.create({
   */
   handleResponse(status) {
     if (status === 401) {
-      this.get('session').invalidate();
-      return true;
+      if (this.get('session.isAuthenticated')) {
+        this.get('session').invalidate();
+      }
+      return new DS.InvalidError(['Session is not authenticated']);
```

Comment:
I don't think this should return an `InvalidError` as a 401 response doesn't indicate that the record is invalid but that the request wasn't allowed due to missing authorization or insufficient rights. When returning an `InvalidError` Ember Data would actually try to get errors for the record from the error which isn't really applicable in this case as the error doesn't apply to the record but to the request as such.